### PR TITLE
feat: insecure http password and weak encryption

### PR DESCRIPTION
**Pull request by vjerci**
## Description
This pr adds insecure http password rule for rails
And enriches weak encryption rule for ruby

## Checklist

- [x] I've added test coverage that shows my fix or feature works as expected.
- [x] I've updated or added documentation if required.
- [x] I've included usage information in the description if CLI behavior was updated or added.
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

File: integration/rules/ruby_test.go [hunk 1]
```diff
@@ -70,6 +70,10 @@ func TestRubyRailsInsecureCommunication(t *testing.T) {
 	getRunner(t).runTest(t, rubyRulesPath+"rails/insecure_communication")
 }
 
+func TestRubyRailsInsecureHTTPPassowrd(t *testing.T) {
```
Comment (elsapet) [inline integration/rules/ruby_test.go:99]:
```suggestion
func TestRubyRailsInsecureHTTPPassword(t *testing.T) {
```

File: pkg/commands/process/settings/rules/ruby/lang/weak_encryption.yml [hunk 2]
```diff
@@ -34,6 +34,11 @@ patterns:
           - export
           - to_pem
           - to_s
+  - pattern: |
+      OpenSSL::PKey::RSA.new($<LENGTH>)
```
Comment (elsapet) [inline pkg/commands/process/settings/rules/ruby/lang/weak_encryption.yml]: Do we want DSA here too? 

https://ruby-doc.org/stdlib-3.1.1/libdoc/openssl/rdoc/OpenSSL/PKey/DSA.html

File: pkg/commands/process/settings/rules/ruby/lang/weak_encryption/testdata/openssl_dsa.rb [hunk 3]
```diff
@@ -1,5 +1,11 @@
+size = 2048
 cipher = OpenSSL::Cipher.new('aes-128-cbc')
-dsa_encrypt = OpenSSL::PKey::DSA.new(2048)
+rsa_encrypt = OpenSSL::PKey::RSA.new(size)
```
Comment (elsapet) [inline pkg/commands/process/settings/rules/ruby/lang/weak_encryption/testdata/openssl_dsa.rb]: We'd want these changes in the RSA file: https://github.com/Bearer/bearer/blob/main/pkg/commands/process/settings/rules/ruby/lang/weak_encryption/testdata/openssl_rsa.rb

**Review by elsapet — COMMENTED** [review]
I think we can support both RSA and DSA libraries here

File: pkg/commands/process/settings/rules/ruby/lang/weak_encryption.yml (hunk 2 above)
Comment (vjerci) [reply pkg/commands/process/settings/rules/ruby/lang/weak_encryption.yml]: Had to do some googling to figure it out,  
found some resource that said DSA 1024 are secure enough.

File: pkg/commands/process/settings/rules/ruby/lang/weak_encryption/testdata/openssl_dsa.rb (hunk 3 above)
Comment (vjerci) [reply pkg/commands/process/settings/rules/ruby/lang/weak_encryption/testdata/openssl_dsa.rb]: nice catch

**Review by elsapet — APPROVED** [review]

**Pull request merged**